### PR TITLE
Chunking for `GreedyScheduler`

### PR DESCRIPTION
**Pull request by carstenbauer**
Quick and dirty attempt at #66

(only very technically breaking because of the new type parameter)

### CODE REVIEW

File: src/schedulers.jl [hunk 1]
```diff
@@ -204,20 +204,74 @@ some additional overhead.
...
+    if !isnothing(nchunks) || !isnothing(chunksize)
+        chunking = true
+    end
+    if !chunking
+        nchunks = -1
+        chunksize = -1
+    else
+        # only choose nchunks default if chunksize hasn't been specified
+        if isnothing(nchunks) && isnothing(chunksize)
+            nchunks = 4 * nthreads(:default)
```
Comment (MasonProtter) [inline src/schedulers.jl:260]: I wonder if this is too small. There's not many opportunities for greediness if each task only gets around 4 chunks to process.

Comment (carstenbauer) [reply src/schedulers.jl:260]: Yeah we might want to test and tune this.

Comment (carstenbauer) [reply src/schedulers.jl]: I've changed it to `10*nthreads()`

**Comment by carstenbauer** [comment]
I plan to merge this soon.

**Pull request merged**